### ISSUE TITLE
npmjs.com description of compatible session stores garbled

### ISSUE THREAD

**Issue by chojrak11** [expressjs/session#248]
https://www.npmjs.com/package/express-session#compatible-session-stores
There are markdown-like tags instead of sensible display.

**Comment by dougwilson** [comment]
Hi! AFAIK, that was working just fine before when I published and nothing has changed. Perhaps it is a bug in their markdown rendering? Can you file a bug against their website, please?

**Comment by cianclarke** [comment]
@dougwilson @chojrak11 Done some digging on this, here's what I've found:

- npm/newww uses npm/marky-markdown to parse it's markdown. npm/marky-markdown uses markdown-it/markdown-it in "default mode" to parse markdown. 
- [Here's a testcase](https://gist.github.com/cianclarke/1b89969d0446bb367378) of feeding a reduced testcase of this repo's `README.md` into markdown-it/markdown-it. Note the expected and actual result. 
- However, I've also fed the example into the [CommonMark Dingus](http://spec.commonmark.org/dingus/) and found the output produces a result as observed in `actual.html` - so not sure what to make of this..
- This seems to be caused by the lack of a newline separating the reference definition from the usage of said reference. I can't find the specific point in the [CommonMark definition for references](http://spec.commonmark.org/0.23/#link-reference-definitions) which explicitly enforces this behavior, but note all their examples include this newline whitespace. 
Maybe my grep is broken.. Other eyes on this document might help!  

**tl;dr**: It looks like this Markdown is more than likely not to spec, but I'm not 100% sure. You can fix this issue by moving all your "Compatible Session Stores" references to the bottom of the file, or by introducing whitespace. I've done this [here](https://github.com/expressjs/session/pull/280), but maybe you'd rather get to the bottom of the "is this to spec" question - your call :-)

**Comment by dougwilson** [comment]
Wow, @cianclarke, thank you so much for taking the time to dig in and look! And even more thanks for the pull request :)!

**Comment by revin** [comment]
Hi everyone, we're tracking this in marky-markdown; I just wanted to point out that the part of the CommonMark spec that I think is causing this behavior is example 166 ([172](http://spec.commonmark.org/0.24/#example-172) in the latest 0.24 version of the spec), where it says, "a link reference definition cannot interrupt a paragraph".

That said, our goal is to be able to render anything to github can render, so I'm going to look at relaxing that rule in our implementation.

Thanks!

**Comment by dougwilson** [comment]
Awesome, thanks, @revin!

**Comment by cianclarke** [comment]
@dougwilson No problem!
Thanks @revin for pointing out the behavior in CommonMark Spec defining why this is wrong - my spec-grep was indeed broken, hah!
